File: PYTHON/Pure Functions/is armstrong.py
def is_armstrong (num):
    num = int(num)
    orig = int(num)
    dig = digit_counter(num)
    sums = int(0)
    while (num > 0):
        temp = int(num % 10)
        temp = int(pow(temp, dig))
        sums = int(sums + temp)
        num = int(num / 10)
    if (sums == orig):
        return 1
    else:
        return 0

def digit_counter (num):
    num = int(num)
    counter = int(0)
    if (num == 0):
        counter = int(1)
    else:
        while (num > 0):
            num = int(num / 10)
            counter = int(counter + 1)
    return counter

x = input("User, enter a number\n")
x = int(x)

results = is_armstrong(x)
if (results == 1):
    print("Is an Armstrong")
else:
    print("Is NOT an Armstrong")
input("\nDONE!\n")

File: C#/code/code_functions/is_armstrong.cs
using System;
using static System.Math;
namespace aviv_yunker {
	class is_armstrong {
		public static void Main (string [] args) {
			string x;
			int num, results;
			System.Console.WriteLine("User, enter a number: ");
			x = System.Console.ReadLine();
			num = int.Parse(x);
			results = is_num_armstrong(num);
			if (results == 1) {
				System.Console.WriteLine("{0} IS an Armstrong", num);
			}
			else {
				System.Console.WriteLine("{0} is NOT an Armstrong", num);
			}
		}
		public static int is_num_armstrong (int num) {
			int orig = num;
			int d = digit_counter(num);
			int temp, sums = 0;
			while (num > 0) {
				temp = num % 10;
				temp = (int)Math.Pow(temp,d);
				sums += temp;
				num /= 10;
			}
			if (sums == orig) {
				return 1;
			}
			else {
				return 0;
			}
		}
		public static int digit_counter (int num) {
			int counter = 0;
			if (num == 0) {
				counter = 1;
			}
			else {
				while (num > 0) {
					counter++;
					num /= 10;
				}
			}
			return counter;
		}
	}
}
